crates/trie/common/src/hashed_state.rs

```diff
@@ -198,6 +198,50 @@ impl HashedPostState {
         targets
     }
 
+    /// Partition the state update into two state updates:
+    /// - First with accounts and storages slots that are present in the provided targets.
+    /// - Second with all other.
+    pub fn partition_by_targets(mut self, targets: &MultiProofTargets) -> (Self, Self) {
+        let mut state_updates_not_in_targets = Self::default();
+
+        self.accounts.retain(|&address, account| {
+            if targets.contains_key(&address) {
+                return true
+            }
+
+            state_updates_not_in_targets.accounts.insert(address, *account);
+            false
+        });
+        self.storages.retain(|&address, storage| {
+            let fetched_targets = targets.get(&address);
+            let fetched_storage_state_update =
+                state_updates_not_in_targets.storages.entry(address).or_default();
+
+            if fetched_targets.is_none() {
+                fetched_storage_state_update.wiped = storage.wiped;
+            }
```

Comment:
unclear why we always initialize a storage entry for account even if it has been fetched, fix or docs pls

Reply:
fixed and made it more explicit